gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/versions/RegionVersionVector.java

```diff
@@ -800,6 +797,70 @@ public void recordVersionForSnapshot(T member, long version, EntryEventImpl even
     }
   }
 
+  public Map<T, RegionVersionHolder<T>> getMemToVSnapshotCopy() {
+    Map<T, RegionVersionHolder<T>> m = new HashMap();
+    m.putAll(((CopyOnWriteHashMap)memberToVersionSnapshot).getInnerMap());
+    for (Map.Entry<T, RegionVersionHolder<T>> entry : m.entrySet()) {
+      m.put(entry.getKey(), entry.getValue().clone());
+    }
+    return m;
+  }
+
+  public Map<T, RegionVersionHolder<T>> recordVersionForSnapshotWithoutPublish(T member, long version,
+                                                                               Map<T, RegionVersionHolder<T>> m) {
+    LogWriterI18n logger = getLoggerI18n();
+    T mbr = member;
+    GemFireCacheImpl cache = GemFireCacheImpl.getInstance();
+    if (cache != null && !cache.snapshotEnabled()) {
+      return null;
+    }
+    RegionVersionHolder<T> holder;
+    Map<T, RegionVersionHolder<T>> forPrinting;
+    //Find the version holder object
+    synchronized (memberToVersionSnapshot) {
+      holder = m.get(mbr);
+      if (holder == null) {
+        mbr = getCanonicalId(mbr);
+        holder = new RegionVersionHolder<T>(mbr);
+      }
+      holder.recordVersion(version, logger);
+
+      // instead of putting the holder save it
+      // and do a putAll
+      m.put(holder.id, holder);
+      forPrinting = m;
```

Comment:
Do we need this forPriniting map. I see it is being assigned with 'm'. Why can't we use 'm' only?